File: src/modules/company/validators/company_delete_validator.py
from src.modules.company.models import Company
from src.modules.user.models import User
from src.modules.role.models import Role
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.exceptions.conflict_exception import ConflictException


def company_delete_validator(company_id: int):

    company = Company.objects.filter(id=company_id, deleted_at=None).first()

    if company is None:
        raise NotFoundException("Empresa não encontrada!")

    has_users = User.objects.filter(company__id=company_id, deleted_at=None).first()

    if has_users is not None:
        raise ConflictException(
            "Não foi possível deletar empresa por que possui usuários vinculados!"
        )

    has_roles = Role.objects.filter(company__id=company_id, deleted_at=None).first()

    if has_roles is not None:
        raise ConflictException(
            "Não foi possível deletar empresa por que possui perfis vinculados!"
        )

File: src/modules/company/validators/company-delete.validator.ts
import { prisma } from '../../../infra/database/prisma.service';
import {
  ConflictException,
  NotFoundException,
} from '../../../shared/exceptions';

export async function companyDeleteValidator(companyId: number) {
  const company = await prisma.company.findFirst({
    where: {
      id: companyId,
      deletedAt: null,
    },
  });

  if (!company) throw new NotFoundException('Empresa não encontrada!');

  const hasUsers = await prisma.user.findFirst({
    where: {
      companyId: companyId,
      deletedAt: null,
    },
  });

  if (hasUsers)
    throw new ConflictException(
      'Não foi possível deletar empresa por que possui usuários vinculados!',
    );

  const hasRoles = await prisma.role.findFirst({
    where: {
      companyId: companyId,
      deletedAt: null,
    },
  });

  if (hasRoles)
    throw new ConflictException(
      'Não foi possível deletar empresa por que possui perfis vinculados!',
    );
}
